src/Magnum/ImGuiIntegration/Integration.cpp

```diff
@@ -0,0 +1,305 @@
+        const ImDrawList* cmdList = drawData->CmdLists[n];
+        ImDrawIdx indexBufferOffset = 0;
+
+        _vertexBuffer.setData(
+                {cmdList->VtxBuffer.Data, std::size_t(cmdList->VtxBuffer.Size)},
+                GL::BufferUsage::StreamDraw);
+        _indexBuffer.setData(
+                {cmdList->IdxBuffer.Data, std::size_t(cmdList->IdxBuffer.Size)},
+                GL::BufferUsage::StreamDraw);
+
+        for(int c = 0; c < cmdList->CmdBuffer.Size; ++c) {
+            const ImDrawCmd* pcmd = &cmdList->CmdBuffer[c];
+
+            auto userTexture = static_cast<GL::Texture2D*>(pcmd->TextureId);
+            _shader.bindTexture(userTexture ? *userTexture : _texture);
+
+            GL::Renderer::setScissor({
+                    {int(pcmd->ClipRect.x), fbHeight - int(pcmd->ClipRect.w)},
+                    {int(pcmd->ClipRect.z), fbHeight - int(pcmd->ClipRect.y)}});
+
+            _mesh.setCount(pcmd->ElemCount);
+            _mesh.setIndexBuffer(_indexBuffer, indexBufferOffset*sizeof(ImDrawIdx),
+                    sizeof(ImDrawIdx) == 2
+                    ? GL::MeshIndexType::UnsignedShort
+                    : GL::MeshIndexType::UnsignedInt);
+
+            indexBufferOffset += pcmd->ElemCount;
+
+            _mesh.draw(_shader);
+        }
+    }
+
+    GL::Renderer::disable(GL::Renderer::Feature::ScissorTest);
+    GL::Renderer::enable(GL::Renderer::Feature::FaceCulling);
+    GL::Renderer::enable(GL::Renderer::Feature::DepthTest);
+}
+
+namespace Implementation {
+
+ImGuiShader::ImGuiShader() {
+    #ifdef MAGNUM_IMGUIINTEGRATION_BUILD_STATIC
+    /* Import resources on static build, if not already */
+    if(!Utility::Resource::hasGroup("MagnumImGuiIntegrationShaders"))
+        importShaderResources();
+    #endif
+    Utility::Resource rs("MagnumImGuiIntegrationShaders");
+
+#ifndef MAGNUM_TARGET_GLES
+    const GL::Version version = GL::Context::current().supportedVersion({GL::Version::GL330, GL::Version::GL310, GL::Version::GL300, GL::Version::GL210});
+#else
+    const GL::Version version = GL::Context::current().supportedVersion({GL::Version::GLES300, GL::Version::GLES200});
+#endif
+
+    GL::Shader vert{version, GL::Shader::Type::Vertex};
+    GL::Shader frag{version, GL::Shader::Type::Fragment};
+
+#ifndef MAGNUM_TARGET_GLES
+    if(version != GL::Version::GL210) {
+        vert.addSource({"#define NEW_GLSL\n"});
+        frag.addSource({"#define NEW_GLSL\n"});
+    }
+
+    // Enable explicit attribute locations in shader if they are supported, otherwise bind locations directly
+    if(GL::Context::current().isExtensionSupported<GL::Extensions::ARB::explicit_attrib_location>(version)) {
+        vert.addSource({"#define EXPLICIT_ATTRIB_LOCATION\n"});
+    } else {
+        bindAttributeLocation(Position::Location, "Position");
+        bindAttributeLocation(TextureCoordinates::Location, "UV");
+        bindAttributeLocation(Color::Location, "Color");
```

Comment:
At first I thought imgui requires the variables to be named like this, but apparently not. In that case, can these follow the naming scheme used in other shaders (in particular, lowercase first letter)? Thank you!

Reply:
Yes, will do! Forgot about that 👍